### DISCUSSION TITLE
WindowBuilder ScrollBar Bug

### DISCUSSION THREAD

**Discussion by shadowxdgamer** [eclipse/windowbuilder#695] [General]
There is a problem with WindowBuilder and it halted my work
when ever i use the scroll bar to move around the mousedrag feature to drag elements or to resize them stops working well i can move them with the arrow keys that works but i can't resize
Here is a video for better explanation :

https://github.com/eclipse-windowbuilder/windowbuilder/assets/64868173/97bdf880-3128-4500-96ea-7faf76f79fdf

I'm using JDK17 and latest version of Eclipse
is there a workaround to fix this? Thanks for answering

**Comment by ptziegler** [comment]
Thanks for bringing this up. There's not a workaround/fix yet but I'll try to have a look at this within the next few days.

**Comment by ptziegler** [comment]
@shadowxdgamer I've just merged #699 which should fix this problem. Given that this problem is blocking your current task, I suggest installing upcoming nightly build via:
https://download.eclipse.org/windowbuilder/updates/nightly/latest/

**Comment by shadowxdgamer** [comment]
@ptziegler Thanks for the reply the implemented fix works like a charm but the nightly build has a problem, you can't add items directly from the palette (when you choose an item it becomes selected)
![image](https://github.com/eclipse-windowbuilder/windowbuilder/assets/64868173/8303caa6-a9fd-4b48-a5fb-b13786eed40a)
I will wait for the latest version of WindowBuilder as for my work i will try to minimize the size of my work area so I don't have to use the scrollbars
Thanks for making such a helpful tool.

**Reply by ptziegler** [reply]
> you can't add items directly from the palette (when you choose an item it becomes selected)

Can you elaborate on that? The item becomes selected, yes. But if you hover over the design page, the item should show up and you should be able to just drop it.

[output.webm](https://github.com/eclipse-windowbuilder/windowbuilder/assets/70652978/9ea24fd6-572d-4d76-8b8b-bf6681c6ba9a)

**Comment by shadowxdgamer** [comment]
I don't know why I have the problem but I can only select them and it doesn't allow me to put the item in the panel

https://github.com/eclipse-windowbuilder/windowbuilder/assets/64868173/46296afe-263f-41df-8749-0b1bcfaf02ad

**Reply by ptziegler** [reply]
You're right... The problem shows up on Windows but not on Linux, starting with the 1.15.0.
For some reason, pressing the mouse is registered, but releasing it is not. What a mess!

**Reply by ptziegler** [reply]
I have isolated 360a0a678fb52b7a8e33b5fbb97de7967e6cc069 as the offending commit.

**Reply by ptziegler** [reply]
Because I removed the `setCapture() ` method from Figure, the palette is now using `setCapture` from the SWT Control class, which has the same signature. Argh!

https://github.com/eclipse-windowbuilder/windowbuilder/blob/c466e3deaa4d505d43d814ea7a9ce274053001cf/org.eclipse.wb.core/src/org/eclipse/wb/core/controls/palette/PaletteComposite.java#L425-L434

The reason this problem doesn't show up on Linux is because `setCapture()` only has a dummy implementation...
```
public void setCapture (boolean capture) {
	checkWidget();
	/* FIXME !!!!! */
	/*
	if (capture) {
		OS.gtk_widget_grab_focus (handle);
	} else {
		OS.gtk_widget_grab_default (handle);
	}
	*/
}
```

**Reply by ptziegler** [reply]
I've merged #704, which should fix this selection bug. @shadowxdgamer Can you try out the M2 and see if this is also resolved on your side?
https://download.eclipse.org/windowbuilder/updates/milestone/S202401292103

**Reply by shadowxdgamer** [reply]
Thank you the new update, it works perfectly fine.
Thank you for your hard work Senior dev.

**Reply by ptziegler** [reply]
That's good to hear! 😄 Thank you for taking the time and testing those fixes.
